### ISSUE TITLE
Investigate Hierarchical Layering

### ISSUE THREAD

**Issue by MadBomber** [MadBomber/justprep#9]
Consider a directory tree that looks like this:

```
home
├── main.just
└── projects
    ├── main.just
    ├── thing_one
    │   └── main.just
    └── thing_two
        └── main.just
```

Now suppose you have CD'ed into your thing_two project.  What you may have is an explicit inclusion of all of the upper-level main.just files so that at the lowest level all the defined recipes are available.  BUT if there was an implicit hierarchical layering in effect the upper level main.just files would be implicitly included.

However, sometimes implicit inclusion can lead to trouble.  So is it worth considering?

With implicit layering you have access to all recipes that have been defined in the directory path with their lower-level over-rides.

You have the ability to do the same thing currently with explicit inclusions.  A potential problem occurs when you move a directory from one path tree to another.  But because of the explicit nature of the inclusions, the justfiles generated will include recipes that may no longer be appropriate.

**Comment by MadBomber** [comment]
With v1.2.2 path globing capability you can get this kind of functionality.

For example is you have a project(s) root directory of /projects you can use

  include /projects/**/*.just

To load in all of the just recipies starting at the projects root down to what ever level in which you are currently working.

**Comment by MadBomber** [comment]
implemented with v1.2.2
